### ISSUE TITLE
SAS kernel does not work on Juypter - local host

### ISSUE THREAD

**Issue by TomGauss** [sassoftware/saspy#93]
I tried so hard to make the juypter talk to SAS. However, I failed in the last step. Can anyone help me with it. The R kernel and Python kernel works fine on Juypter. The error is:

**The application could not log on to the server. The server process did not start.
SAS process has terminated unexpectedly. RC from wait was: 4294967290
SAS Connection failed. No connection established. Double check you settings in sascfg.py file.**
Full error screenshot enter image description here

This is my setting in the sascfg.py

cpW  =  "D:\\Program Files\\SASHome\\SASDeploymentManager\\9.4\\products\\deploywiz__94250__prt__xx__sp0__1\\deploywiz\\sas.svc.connection.jar"
cpW += ";D:\\Program Files\\SASHome\\SASDeploymentManager\\9.4\\products\\deploywiz__94250__prt__xx__sp0__1\\deploywiz\\log4j.jar"
cpW += ";D:\\Program Files\\SASHome\\SASDeploymentManager\\9.4\\products\\deploywiz__94250__prt__xx__sp0__1\\deploywiz\\sas.security.sspi.jar"
cpW += ";D:\\Program Files\\SASHome\\SASDeploymentManager\\9.4\\products\\deploywiz__94250__prt__xx__sp0__1\\deploywiz\\sas.core.jar"
cpW += ";D:\\ProgramData\\Miniconda3\\Lib\\site-packages\\saspy\\java\\saspyiom.jar"

winlocal = {'java'      : 'java',
            'encoding'  : 'windows-1252',
            'classpath' : cpW
            }

Same question I have posed on stackoverflow:
https://stackoverflow.com/questions/48399232/sas-kernel-does-not-work-on-juypter

**Comment by tomweber-sas** [comment]
Ok, so in a python kernel, can you successfully run the following? Can you show the code and results.

```
import saspy
sas = saspy.SASsession(cfgname='winlocal')
sas
cars = sas.sasdata('cars', 'sashelp')
cars.head()
```
Also, do you have the path to the sspiauth.dll in your System Path? Can you confirm that?

Thanks,
Tom

**Comment by TomGauss** [comment]
Hi tomweber, 

  Thanks very much for reply and attention. 

   In python kerner, the sas cannot run. see the description below. I also attached the PDF file.

   sspiauth.dll is in my system path. I double checked the system variable, and try to make sure the command pump can access the file. See the picture in file

![sspiauth dll](https://user-images.githubusercontent.com/32924623/35391260-0eff6e1c-01d6-11e8-8913-a87d7fe23755.png)
. 

   
[Error message in python kernel.pdf](https://github.com/sassoftware/saspy/files/1664186/Error.message.in.python.kernel.pdf)

: 

`The application could not log on to the server. The server process did not start.
SAS process has terminated unexpectedly. RC from wait was: 4294967290
SAS Connection failed. No connection established. Double check you settings in sascfg.py file.

Attempted to run program java with the following parameters:['java', '-classpath', 'D:/Program Files/SASHome/SASDeploymentManager/9.4/products/deploywiz__94250__prt__xx__sp0__1/deploywiz/sas.svc.connection.jar;D:/Program Files/SASHome/SASDeploymentManager/9.4/products/deploywiz__94250__prt__xx__sp0__1/deploywiz/log4j.jar;D:/Program Files/SASHome/SASDeploymentManager/9.4/products/deploywiz__94250__prt__xx__sp0__1/deploywiz/sas.security.sspi.jar;D:/Program Files/SASHome/SASDeploymentManager/9.4/products/deploywiz__94250__prt__xx__sp0__1/deploywiz/sas.core.jar;D:/ProgramData/Miniconda3/Lib/site-packages/saspy/java/saspyiom.jar', 'pyiom.saspy2j', '-host', 'localhost', '-stdinport', '64810', '-stdoutport', '64811', '-stderrport', '64812', '-zero', '']

Be sure the path to sspiauth.dll is in your System PATH

No SAS process attached. SAS process has terminated unexpectedly.
No SAS process attached. SAS process has terminated unexpectedly.
No SAS process attached. SAS process has terminated unexpectedly.
No SAS process attached. SAS process has terminated unexpectedly.
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-1-c6cb007830d1> in <module>()
      2 sas = saspy.SASsession(cfgname='winlocal')
      3 sas
----> 4 cars = sas.sasdata('cars', 'sashelp')
      5 cars.head()

d:\programdata\miniconda3\lib\site-packages\saspy\sasbase.py in sasdata(self, table, libref, results, dsopts)
    505             results = self.results
    506         sd = SASdata(self, libref, table, results, dsopts)
--> 507         if not self.exist(sd.table, sd.libref):
    508             if not self.batch:
    509                 print(

d:\programdata\miniconda3\lib\site-packages\saspy\sasbase.py in exist(self, table, libref)
    397         :rtype: bool
    398         """
--> 399         return self._io.exist(table, libref)
    400 
    401     def sasets(self) -> 'SASets':

d:\programdata\miniconda3\lib\site-packages\saspy\sasioiom.py in exist(self, table, libref)
    982       l2 = ll['LOG'].rpartition("TABLE_EXISTS= ")
    983       l2 = l2[2].partition("\n")
--> 984       exists = int(l2[0])
    985 
    986       return exists

ValueError: invalid literal for int() with base 10: 'No SAS process attached. SAS process has terminated unexpectedly.'`

Best regards
Tom

**Comment by tomweber-sas** [comment]
Ok, thanks, so this isn't really a sas_kernel issue, saspy can't connect to your SAS yet, so we have to start there. What SAS do you have installed? Is it Base SAS or just a client, like EG? I've ggoled the error message, as that's not one I've seen before. Most issues I see are all with clients trying to connect to a remote IOM server of some kind.
Thsi Local connection type is to connect to a Base SAS deployment. If what you have in D:\... isn't at least a Base SAS deployment, that would certainly be a problem. Do you have sas.exe in there?
BTW, which version of saspy do you have? Didn't get that in the ouput; submitting 'sas' would have printed out info, but sine the sasdata() was in the same cell, Jupyter didn't get that output written.

Thanks,
Tom

**Comment by TomGauss** [comment]
Hi tomweber,

   Thanks very much for help me this. 

    What SAS do you have installed?

     I installed SAS 9.4 and it's not just a client. More specifically
![sas](https://user-images.githubusercontent.com/32924623/35392955-77cd3cda-01db-11e8-999f-9903e4e15ffc.png)

     Do you have sas.exe in there?
     Yes I did. Actually, the local SAS runs pretty well. I have used it and it's completely fine. 

     which version of saspy do you have?
     The version I use is below:
     SAS-kernel==2.1.7
     saspy==2.2.1
    
     I also re-run the code you sent me before, and here is the result:
![123](https://user-images.githubusercontent.com/32924623/35393341-99964824-01dc-11e8-8016-1c90e7881602.JPG)

    My suspicious:
       Would you think this could be due to some reasons of Java? 
       I am thinking this might be due to the fact that java cannot open SAS file correctly? 
       However, I don't konw how to test this, 
       I am actually a statistic programmer, I am heavy lack of knowledge of other language. 

   Thanks for your help. 

Best regards
Tom

**Comment by tomweber-sas** [comment]
Ok, so that looks like it should be able to work then. I haven't seen this error before. What version of java do you have? I don't see anything here to make me suspicious of java though. 
I wonder if it could be a permission thing? I take it you're only running with your id in all cases? Running SAS itself and saspy; well, actually, you're running jupyter, are you starting it us w/ your id, or is it running with some other permissions??? 
Can you just bring up a dos shell and run python natively (what version of pythong BTW?) and submit the import, sassession; sas  and see what we get?
I'll have to dig in here some too to see how this error could happen with base SAS.

We'll get it figured out! 
Thanks,
Tom

**Comment by TomGauss** [comment]
What version of java do you have?
I am using version 8 update 161. 
![111](https://user-images.githubusercontent.com/32924623/35394417-a453e048-01df-11e8-8093-eddf3d42b7fe.JPG)

 I take it you're only running with your id in all cases? 
 Yes, lucky for me. I am allowed to use my personal PC in my work. I am the administrator of the laptop, and there is no such thing about other admistrator account or control.  You can regard the computer as a pure personal PC. 

 Running SAS itself, jupyter and saspy.
 The way I run SAS is simple click SAS.exe. then there would a user panel control and ask me to click yes. 
 The way I run jupyter is simple using "jupyter notebook" in comman prompt. See the picture below, poten is my ID. I do not need to imput my id to run the jupyter in the command line, or should I try using it?
![222](https://user-images.githubusercontent.com/32924623/35394802-97520cfc-01e0-11e8-9f2e-ac49824c7792.JPG)

Can you just bring up a dos shell and run python natively (what version of pythong BTW?) and submit the import, sassession; sas and see what we get?
  My python version is 3.6.3
  Unfortunately, it still fails. see the picture below:
![3333](https://user-images.githubusercontent.com/32924623/35395038-24ce3f60-01e1-11e8-9dee-440324106b7e.JPG)

The reason I am suspicious about Java,
   I tried edit the absolute path for java and change it back. Everytime I change the java path and run SAS code in Jupyter, it would take some time. Then there would be a network access permission pop out, and ask me to agree. I agreed every time, but the SAS still fails. This is the reason I suspect Java. 

Thanks for help. I actully also googled the whole website, and haven't found anyone has the similar problem as me. I am kind of desperate, I am so glad you can see this. 

Best regards
Yan

**Comment by tomweber-sas** [comment]
So, I don't see anything out of the ordinary w/ your java, SAS, python, saspy or saspy config. This all should just be working. So, there's got to be something unusual about your environment. I'm checking w/ IOM folks here about how you could be getting this error trying to make a local connection. And, the network pop up you're getting is also a suspicious clue too. I don't suspect java itself. This is a personal machine on your corporate network? And the fact that this error only 'seems' (don't know enough yet) to be associated with remote IOM connections, if odd. This 'should' all just be localhost, but there seems to be something else going on here. 

Ok, got an idea from one of our guys. So, in the windows registry is the command to start SAS for this local IOM connection I'm trying to do. I've seen a problem a few times where a user doesn't have this in their registry and they get this error:
The application could not find a command to launch a SAS Workspace Server.

In the troubleshooting section of the doc, this is documented with the workaround. This isn't what you're getting, but the thought is, if you have a valid command there that is being used, but it isn't (effectively) sas.exe, but rather something that is trying to access something else (remote connection), then maybe that could be why you're getting some network issue and an error that seems to be associated w/ a remote connection to somewhere.

So, lets try the following and see what happens. Add the following key to your winlocal config definition and edit the correct path to your sas.exe, then give it a try again; python shell (less other pieces for now).
This will override whatever might be in the registry and should start your local SAS.

```
'javaparms' : ['-Dcom.sas.iom.orb.brg.zeroConfigWorkspaceServer.sascmd=C:\PROGRA~1\SASHome\SASFOU~1\9.4\SAS.EXE']
```
I'll track down what the key is in the registry so we can look to see what's there too.

**Comment by TomGauss** [comment]
Hi tomweber,

   Thanks for reply. 

   Now it seems something wrong here, because I cannot load saspy. see what happend in picture
![1](https://user-images.githubusercontent.com/32924623/35398705-811d50b8-01ea-11e8-9ff9-4c98d741f03d.JPG)
in my winlocal config:
![2](https://user-images.githubusercontent.com/32924623/35398724-8a1ecf5c-01ea-11e8-8c09-b02dc448bebc.JPG)

   Jupytor also says the SAS kernel is dead and can't be restart. 

   Any further advice? or am I mistaken on something here. 

Best regards
Tom

**Comment by tomweber-sas** [comment]
you need to add a ',' comma in your config definition to the line above this. That;s all.

**Comment by tomweber-sas** [comment]
Also, just in case. every example, and in my own install, the path they use is always the goofy ~ 8char windows paths (to remove spaces from w/in the path which half the system can't handle on windows), so unless this just works, you might want to try the ~ version.
```
'javaparms' : ['-Dcom.sas.iom.orb.brg.zeroConfigWorkspaceServer.sascmd=D:\PROGRA~1\SASHome\SASFOU~1\9.4\SAS.EXE']
```

**Comment by TomGauss** [comment]
Hi tomweber,

  Thanks for reply. I corrected the code and also tried the ~ version.

  It still mentions fail like before
![000000](https://user-images.githubusercontent.com/32924623/35399693-c297c940-01ec-11e8-999a-1bdf97340df9.JPG)

any clue ?

Best regards
Tom

**Comment by tomweber-sas** [comment]
I tracked down the registry key. it's
HKEY_CLASSES_ROOT\CLSID\{440196D4-90F0-11D0-9F41-00A024BB830C}\LocalServer32
Can you see what you have there?
you can use 'regedit' to nevigate to the key and see the value.

**Comment by TomGauss** [comment]
yes, I have something here
![33333333333](https://user-images.githubusercontent.com/32924623/35400109-d7a7c7ee-01ed-11e8-9ca4-e18358087586.JPG)

The value name is
D:\PROGRA~2\SASHome\SASFOU~1\9.4\SAS.EXE -config "D:\Program Files\SASHome\SASFoundation\9.4\sasv9.cfg"  -objectserver -nologo -noterminal -noprngetlist

**Comment by tomweber-sas** [comment]
well, that looks correct. D: is a physical drive on your PC, right? Not a mounted drive (network)?
Just checking. I'm not seeing any reason for this error. Is there anything else you can think of that might be relevant? I'm at a loss.

**Comment by TomGauss** [comment]
Hi tomweber,

   Yes, D: is a physical driver on my PC. 
    I am thinking about I install Anaconda and miniconda at the same time, and saspy is in miniconda, but Python is in Anaconda. Could this screw something in there?

  Thanks for your help. 

Best regards
Tom

**Comment by TomGauss** [comment]
Hi tomweber,

   You might be off from work just like me. I would like to add last piece of information.

   You mentioned before it is suspicious I need to allow the Java to access the network when I first load saspy. Since the session is only a local windows session, why the java need to access the permission to connect to public or private network ? This might be a clue? Also, I am using education network for now. I will give a shot when I am back to apartment. 

   Thanks for all the help so far. Have a good day.

Best regards
Tom

**Comment by TomGauss** [comment]
Hi tomweber,

   I found that the windows defender would ask permission for java to connect network. I disabled the windows defender, however it is still not wokring. Now I suspect the problem might be something in localhost. 
![1234](https://user-images.githubusercontent.com/32924623/35403995-67084700-01f9-11e8-8477-836ec4748da4.JPG)

**Comment by tomweber-sas** [comment]
Well, that's fingers in the pie. How about trying this. Cut-n-paste the command from your windows registry and run that from a dos shell to see what happens. Change the -noterminal to -terminal (just delete the 'no') and it should pop up a window with the log. Like this:
![image](https://user-images.githubusercontent.com/17710182/35404462-13144acc-01d1-11e8-91d4-de6b19bc5e3d.png)

**Comment by TomGauss** [comment]
Hi tomweber,

   I am so glad you are still online. 

   I tried. However, I did not get any window pops out. There is no log or message comes out. It is just the fail information as before. 

Best regards
Tom

**Comment by TomGauss** [comment]
Since the key is sas.py. I decide to upload it here, so you may see anything I missed. I changed it to txt since github does not support .py file

[sascfg.txt](https://github.com/sassoftware/saspy/files/1665387/sascfg.txt)

**Comment by tomweber-sas** [comment]
any chance you can connet up with a webX to look at this in deatil?

**Comment by tomweber-sas** [comment]
I'm not convinces that the path in javaparms is ok; windows doesn't work well with blanks in the path.
'javaparms' : ['-Dcom.sas.iom.orb.brg.zeroConfigWorkspaceServer.sascmd=D:/Program Files/SASHome/SASFoundation/9.4/SAS.EXE']

Can I see the dos windo with output from running the command from your registry key, and how about running it using the path above:  D:/Program Files/SASHome/SASFoundation/9.4/SAS.EXE

Here's one more way to try running it, from powershell:
param([string]$code)
$wks = new-object -comobject SAS.Workspace
$lang=$wks.LanguageService
$lang.Submit($code)
$cc=[int[]]@()
$type=[int[]]@()
$line=[string[]]@()
$lang.FlushLogLines(1024, [ref]$cc, [ref]$type, [ref]$line)
$line
$output = $lang.FlushList(2048) 
$output
$wks.close()

what does that show?

**Comment by TomGauss** [comment]
Hi tomweber, 

   Sorry, I do not understand what webX is. I apology for my poor knoweledge. 

   I changed the path as you mentioned, however still same error message
![1212](https://user-images.githubusercontent.com/32924623/35410265-c52a9bb8-020c-11e8-9cbb-bfcc38dc24b1.JPG)

   I have no idea why I can't run the other way
![321](https://user-images.githubusercontent.com/32924623/35410296-e0dad03a-020c-11e8-85de-69fcec2233ab.JPG)

**Comment by tomweber-sas** [comment]
Hey, webX is a way we can share screens so we can look at this together. It's just over the internet connection to make it easier to look at this at the same time.

That last thing needs to be run in powershell, not a DOS cmd prompt. But I still don't see running the command in the DOS shell.

If you can email me, I can send you a invite to a webX session. If not, we can keep working like this of course.

**Comment by TomGauss** [comment]
It seems powershell still does not accept it. 
![2222222](https://user-images.githubusercontent.com/32924623/35410715-31f015ce-020e-11e8-89d3-4cd3703ebef0.JPG)

**Comment by tomweber-sas** [comment]
ok, and can I see what happens when you try to run the sas command from a DOS shell:

D:\PROGRA2\SASHome\SASFOU1\9.4\SAS.EXE -config "D:\Program Files\SASHome\SASFoundation\9.4\sasv9.cfg" -objectserver -nologo -terminal -noprngetlist

I would cut-n-paste this out of your registry itself, not from above. I don't think my cut-n-paste from your post of this is right.

**Comment by TomGauss** [comment]
Sorry, I forget to include this. 
SAS seems to start succefully?
![3344](https://user-images.githubusercontent.com/32924623/35410980-fce6f5f4-020e-11e8-9c52-f1c59dbd6932.JPG)

**Comment by tomweber-sas** [comment]
Ah, well that worked. Does it work also from DOS? I'll have to get some folks here to look at this and see what they think.

**Comment by TomGauss** [comment]
![3434343434](https://user-images.githubusercontent.com/32924623/35411134-85bd3744-020f-11e8-9635-71aa3af6f41e.JPG)
Yes, it works in dos.

**Comment by TomGauss** [comment]
HI tomweber,

   Thanks very much so far. My remote server also has python, sas and jupyter. I would try to install saspy there, and see what would happen there. 
 
   In this case, I might be able to query my local IT staff members. I'll get back to you if I get any luck. 

Best regards
Tom

**Comment by tomweber-sas** [comment]
ok, well that's good then. There's something else we're missing that's interfering with this somehow then.
Can you look in the Windows event viewer and see if there are any messages that have to do with this? You might want to bring that up and then run a case so you can see if anything shows up then, note the time for the timestamps. Maybe there's an error in there that will clue us in. maybe some other antivirus or firewall or who knows when on the system, is keeping this from working. Like that defender thing with the network.

**Comment by TomGauss** [comment]
Hi tomweber,

  Thanks. I run the case and checked the windows event viewer. Unfortunately, nothing suspecious were found. I only have windows firewall and it is closed. The other thing I have is a CCcleaner. I do not have other anti-virus software. There is no action needed in windows firewall as well. This seems to become a mysterious now.

**Comment by tomweber-sas** [comment]
for the powershell case, I was told you probably need to open powershell 'as administrator', that should 'elevate permissions' and get that case to run. And the same windows guy that gave me that thinks it still might be an issue with the paths in the commands. They've seen issues with that after updates. 

I've been playing around with the command in javaparms myself. When the path isn't good, I do see the same error you are seeing! That's good, actually. I've bee successful running with this (what's in my registry key):

             'javaparms' : ['-Dcom.sas.iom.orb.brg.zeroConfigWorkspaceServer.sascmd="C:\PROGRA~1\SASHome\SASFOU~1\9.4\SAS.EXE" -config "C:\PROGRA~1\SASHome\SASFOU~1\9.4\sasv9.cfg" -objectserver -nologo -noterminal -noprngetlist']

Can you try using this (changed only to get your paths correct), which I think would be this, but double check:

             'javaparms' : ['-Dcom.sas.iom.orb.brg.zeroConfigWorkspaceServer.sascmd="D:\PROGRA~2\SASHome\SASFOU~1\9.4\SAS.EXE" -config "D:\PROGRA~2\SASHome\SASFOU~1\9.4\sasv9.cfg" -objectserver -nologo -noterminal -noprngetlist']

See if that works with your saspy case. You can try it from a DOS shell to be sure it's typed in right:
"D:\PROGRA~2\SASHome\SASFOU~1\9.4\SAS.EXE" -config "D:\PROGRA~2\SASHome\SASFOU~1\9.4\sasv9.cfg" -objectserver -nologo -terminal -noprngetlist
 
I hope that works.

**Comment by tomweber-sas** [comment]
```
"D:\PROGRA~2\SASHome\SASFOU~1\9.4\SAS.EXE" -config "D:\PROGRA~2\SASHome\SASFOU~1\9.4\sasv9.cfg" -objectserver -nologo -noterminal -noprngetlist
```

**Comment by tomweber-sas** [comment]
That (with my paths) worked from DOS, as it does for you, but it also worked from saspy too. It failed with your same error, from saspy, when I didn't have the quotes right or the path wasn't right. So it might just be something with these silly paths in windows. We have a defect open about getting these command paths right in the registry, so it may be something about that tripping it up here.

**Comment by TomGauss** [comment]
![444444](https://user-images.githubusercontent.com/32924623/35412775-2ef8a7c6-0215-11e8-810a-42ccb97cb5c7.JPG)
OMG, you are genius 
It seems work now.

**Comment by TomGauss** [comment]
![66666](https://user-images.githubusercontent.com/32924623/35412864-8f2f203e-0215-11e8-9982-f906fa96867f.JPG)
Jupyter works as well. Excellent work !!!!

**Comment by tomweber-sas** [comment]
If I were genius I'd know what's wrong :)
But, lets try re-regestering your sas command and comment out the javaparms and see if it get's fixed for real.

Apparently that is done by running sas with this option '/regserver'. Like this
sas.exe /regserver

So,
D:\...\sas.exe /regserver 


Run that then comment out javaparms then run fresh python and try saspy

**Comment by tomweber-sas** [comment]
yes, if saspy works, then sas_kernel and jupyter all will work!

**Comment by TomGauss** [comment]
Sorry, I am little lost here. What do I need to run and where? 
Also you are right, if I don't choose the administrtor for my CMD then it would not work as well.

**Comment by tomweber-sas** [comment]
Ah, from DOS, just run    sas.exe /regserver
You might need to fully qualify sas.exe - D:\bla bla\sas.exe /regserver
This should update the command in the refistry key; look and see if it's different after.
Then comment out the javaparms in your saspy config and see if everything works like it should.
Show me what the new key is, also.

**Comment by TomGauss** [comment]
DOS can't read the file path ?

C:\Users\poten
λ D:\Program Files\SASHome\SASFoundation\9.4\SAS.EXE/regserver
'D:\Program' is not recognized as an internal or external command,
operable program or batch file.

**Comment by tomweber-sas** [comment]
of course not, it can't handle blanks in path names! Put quotes aroune the entire "D:\...\sas.exe" /regsrever
or use 
```
D:\PROGRA~2\SASHome\SASFOU~1\9.4\SAS.EXE /regserver
```

**Comment by TomGauss** [comment]
The new register key is:
D:\PROGRA~2\SASHome\SASFOU~1\9.4\SAS.EXE -config "D:\PROGRA~2\SASHome\SASFOU~1\9.4\sasv9.cfg"  -objectserver -nologo -noterminal -noprngetlist
It's different from the old one.  
I think
[sascfg.txt](https://github.com/sassoftware/saspy/files/1665771/sascfg.txt)
 you are right. It's the problem of the path of the register key.  Now it's worked completely. Please accept the fact that you are genius. :)
So in summary the solution is:
  1. fix the path in register key
  2. Run CMD as the admin otherwise it would give the same error as above. 
Thank you very much !!
![56565656](https://user-images.githubusercontent.com/32924623/35413721-4a610514-0218-11e8-8223-f37c2984ce0c.JPG)
![123321](https://user-images.githubusercontent.com/32924623/35413726-4c2f94f0-0218-11e8-9012-9282c676f14e.JPG)

**Comment by tomweber-sas** [comment]
Man alive! We'll I've learned something new today, so I think I'll quit while I'm ahead! :) :)
I'm so glad we got to the bottom of this and you're up and running.
I will add this info to the troubleshooting section of the doc, so it's there for other too!

Thanks!
Tom

**Comment by TomGauss** [comment]
Hey tomweber,

   Yes, I totally agree. I learned a lot of things from this process. Thanks for helping !! Have a good night.

Best regards
Tom

**Comment by tomweber-sas** [comment]
I don't think so, I think something changed about these goofy paths and quotes and things with windows. We have a defect about something having to do with that. That's where I got the /regserver from one of our windows guys. He knew about there being a path issue.
Yes, have a good night and let me know if there is anything else as you start to use saspy and sas_kernel!
Tom

**Comment by tomweber-sas** [comment]
Ok, I've added this to the troubleshooting guide. It's under IOM, in the part with the various IOM Errors, number 5. I will go ahead and close this. If you need to reopen for any reason, no problem. If you run into anything else, just open another issue!

Thanks,
Tom
